Games controller

ilya-frolov opened the pull request:
For review games controller, please.

ilya-frolov committed:
Create GameController:
- add works;
- getAll works;
- delete doesn't work.

ilya-frolov committed:
Create simple CRUD implementation in GameController.

ilya-frolov committed:
Add producer doesn't work.

ilya-frolov committed:
Add producer works.
Add categories doesn't work.

ilya-frolov committed:
Add categories doesn't work.

ilya-frolov committed:
Create CRUD for game with everything except for sales.

ilya-frolov committed:
Create Swagger.

RKashanok committed:
rolled back to boot 2.7.7

ilya-frolov committed:
Merge remote-tracking branch 'origin/dev' into games_controller

# Conflicts:
#	src/main/java/info/freeit/boarderpicker/entity/Game.java

ilya-frolov committed:
Merge remote-tracking branch 'origin/games_controller' into games_controller

# Conflicts:
#	src/main/java/info/freeit/boarderpicker/entity/Category.java
#	src/main/java/info/freeit/boarderpicker/entity/Producer.java
#	src/main/java/info/freeit/boarderpicker/entity/Role.java
#	src/main/java/info/freeit/boarderpicker/entity/Sale.java
#	src/main/java/info/freeit/boarderpicker/entity/User.java

ilya-frolov committed:
1. Delete not usable commented row from application-test.properties.
2. Change version of SpringBoot from 2.7.7 to 2.7.7-SNAPSHOT since the last one didn't work at my end.
3. Change mapping URI in GameController.

ilya-frolov committed:
1. Change all CrudRepositories to JpaRepositories.
2. Move all saving operations from GameController to GameService.
3. Add verification to GameServiceImpl if a game already exists.
4. Change in GameService in methods getGameById(), updateGame() my custom GameNotFoundException to RuntimeException.
5. Add in GameService verification in deleteGame() if Game with the id exists.
6. Add my custom GamesNotFoundException to AppExceptionHandler, but delete "throws GamesNotFoundException" from getAllGames() in GameServiceImpl still not allowed.

ilya-frolov committed:
Add my custom GamesNotFoundException to AppExceptionHandler, but delete "throws GamesNotFoundException" from getAllGames() in GameServiceImpl still not allowed.

ilya-frolov committed:
Delete CategoryService and ProducerService since addressing from GameService now straight to their repositories.

ilya-frolov committed:
Delete throwing IllegalArgumentException from GamesController addGame().

ilya-frolov committed:
Return all entities to the Dev condition.

ilya-frolov committed:
1. Return UserRepository to CrudRepository.
2. Return SpringBoot to version 3.0 -> Change pom.xml
3. Change all imports to Jacarta according to SpringBoot 3 requirements.

ilya-frolov committed:
1. Change 'gameName' to 'name' in Game everywhere and in GameRepository.
2. Delete @Column(name = "gamename") and @Column(name = "picpath") in Game.
3. Create NewGameDto and use it instead of Game in GameController and GameService in addGame() and saveGame() methods.
4. Add @Builder to Producer.
5. Change all logic in saveGame() and updateGame() in GameService.
6. Change 'GAMENAME' to 'NAME' in data.sql and import.sql.
7. Change 'gamename' to 'name' in schema.sql.
8. Create SavedGameDto returning Game with id for future Sale creation.
9. Add logging to GameController and GameService.

RomanKashenok commented on src/main/java/info/freeit/boarderpicker/service/exception/GamesNotFoundException.java:
```diff
@@ -0,0 +1,9 @@
+package info.freeit.boarderpicker.service.exception;
+
+public class GamesNotFoundException extends Exception {
```
давай от рантайма его заэкстендим

RomanKashenok commented on src/main/java/info/freeit/boarderpicker/controller/GameController.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @GetMapping
+    public List<SavedGameDto> getGames() throws GamesNotFoundException {
```
просто GameDto

RomanKashenok commented on src/main/java/info/freeit/boarderpicker/dto/SavedGameDto.java:
```diff
@@ -0,0 +1,39 @@
+    private String name;
+    private String description;
+    private String picPath;
+    private Set<Category> categories = new HashSet<>();
```
давай CategoryDto

ilya-frolov committed:
1. Add logging Hibernate and showing sql tracking to application-test.properties.
2. Delete constructor from Game and Category.
3. Change name PIC_PATH in import.sql.
4. Add category request into import.sql.
5. Delete from GameController log string.
6. Delete from GameService gameBuilder and log string.
7. Add CascadeType.PERSIST to Producer in Game, delete Cascade from Category in Game.
8. Delete data of Game from toString() in Category.
9. Add @JsonIgnore and @EqualsAndHashCode.Exclude to field Game in Category.
10. Delete password for h2 DB from application-test.properties.

iryna-pantsialeyeva committed:
Merge branch 'dev' into games_controller
